src/components/CreateNimi/CreateNimi.tsx

```diff
@@ -130,26 +130,25 @@ export function CreateNimi({ ensAddress, ensName, provider, availableThemes }: C
   const [imgErrorMessage, setImgErrorMessage] = useState('');
   const publishNimiAbortController = useRef<AbortController>();
 
-  console.log('abaliableThemes', availableThemes);
-
   // Form state manager
   const useFormContext = useForm<Nimi>({
     resolver: yupResolver(nimiValidator),
     defaultValues: {
-      displayName: state.displayName || ensName,
-      image: state.image?.url ? state.image : undefined,
-      description: state.description || '',
+      displayName: state.displayName || ensData?.displayName || ensName,
+      image: state.image?.url ? state.image : ensData?.image ? ensData?.image : undefined,
+      description: state.description || ensData?.description || '',
       ensAddress,
       ensName,
       //TODO: Add id-s to links so that it can auto-populate field
-      addresses: [],
-      links: state.links || [],
-      widgets: state.widgets || [
-        {
-          type: NimiWidgetType.POAP,
-        },
-      ],
-      theme: { type: availableThemes.length !== 0 ? availableThemes[0] : NimiThemeType.NIMI },
+      addresses: ensData?.addresses || [],
+      links: state.links || ensData?.links || [],
+      widgets: state.widgets ||
+        ensData?.widgets || [
+          {
+            type: NimiWidgetType.POAP,
+          },
+        ],
+      theme: state.theme || { type: availableThemes.length !== 0 ? availableThemes[0] : NimiThemeType.NIMI },
```

Comment:
This entire section is hard to reason about. Consider:
- not using location state. 
- rename `ensData` in props to `initialNimi` and assign that to `defaultValues`.

Reply:
- I decided to use location state so we don't fetch data twice on the domains section and then again when you click
- Yeah I guess I can rename it but state.something is also an initial value that is prefetched